presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp

```diff
@@ -104,7 +102,9 @@ class PlanNodeSerdeTest : public testing::Test,
     Type::registerSerDe();
     core::PlanNode::registerSerDe();
     core::ITypedExpr::registerSerDe();
-    exec::registerPartitionFunctionSerDe();
+    auto& registry = DeserializationWithContextRegistryForSharedPtr();
```

Comment:
This doesn't look right. 

CC: @kewang1024 @xiaoxmeng

Reply:
Hi Masha @mbasmanova , with exec::registerPartitionFunctionSerDe(), PlanNodeSerdeTest in presto_cpp has two unit test failed with error

"couldn't find key keyChannels in dynamic object"

I got the above piece of logic from Ke, but forgot to include it in the "git commit --amend" in my Velox version advance PR  https://github.com/prestodb/presto/pull/19554. So adding it in this separate PR. This logic fixes the two unit test error on my local.

Reply:
@kewang1024 @xiaoxmeng  Ke, why exec::registerPartitionFunctionSerDe() is not working?

@gggrace14 Ge, what's the error? We should not "manually" register these classes, rather use helper methods like exec::registerPartitionFunctionSerDe().

Reply:
Masha, the error is just

*"couldn't find key keyChannels in dynamic object"*

when we run the partitionAndSerializeNode and shuffleWriteNode tests. It's reproducible locally.

Reply:
So yesterday we merged the refactoring on velox, but haven't started the change on presto native but want to unblock the build

So we have several same spec definition: 
1. In presto native test (Needs to be removed)
2. In velox test (Just removed)
3. In velox prod (Newly introduced)

So this is the 1, and this change would be temporary

Reply:
I reproduced the failure locally. I believe the right fix is to delete HashPartitionFunctionSpec class from this test file (and use the one defined in velox/exec/HashPartitionFunction.h).